src/installer/corehost/cli/deps_entry.cpp

```diff
@@ -21,28 +21,29 @@ static pal::string_t normalize_dir_separator(const pal::string_t& path)
     return normalized_path;
 }
 
-void deps_entry_t::append_resource_path(pal::string_t& base) const
-{
-    assert(asset_type == asset_types::resources);
-
-    pal::string_t pal_relative_path = normalize_dir_separator(asset.relative_path);
-
-    // Resources are represented as "lib/<netstandrd_ver>/<ietf-code>/<ResourceAssemblyName.dll>" in the deps.json.
-    // The <ietf-code> is the "directory" in the pal_relative_path below, so extract it.
-    pal::string_t ietf_dir = get_directory(pal_relative_path);
-    pal::string_t ietf = ietf_dir;
-
-    // get_directory returns with DIR_SEPARATOR appended that we need to remove.
-    remove_trailing_dir_seperator(&ietf);
-
-    // Extract IETF code from "lib/<netstandrd_ver>/<ietf-code>"
-    ietf = get_filename(ietf);
-
-    append_path(&base, ietf.c_str());
-    trace::verbose(_X("Detected a resource asset, will query dir/ietf-tag/resource base: %s asset: %s"), base.c_str(), asset.name.c_str());
-}
-
-bool deps_entry_t::to_path(const pal::string_t& base, bool look_in_base, pal::string_t* str) const
+// -----------------------------------------------------------------------------
+// Given a "base" directory, determine the resolved path this this file.
+//
+// If this file exists within the single-file bundle candidate is:
+//  * If the file was extracted to disk, the full-path to the extracted file.
+//  * Otherwise, the path within the bundle, relative to the "base" directory.
+//    The runtime expects the entries in the TPA,  NativeDllSearchDirectories, etc 
+//    to be absolute paths. Therefore, the relative-paths within the bundle are
+//    expressed as absolute paths with respect to the location of the bundle-file.
```

Comment:
But I agree that this comment is confusing, because it tries to talk about both the current and future state of the machine. I'll fix it to only say that the assembly is expected to be found at the extracted location.

Reply:
So if I read it correctly, this is a point-in-time behavior.
Currently it is expected to actually extract all assemblies on disk. And so we calculate the path as relative to the app-base.
In the future, these will not be extracted and thus the code will change to not include them in TPA (and the logic how they're resolved will have to adapt as well). Right?

Reply:
Yes, this is correct.